prov/efa/src/efa_strerror.c

```diff
@@ -101,18 +125,100 @@ static const char *efa_errno_str(enum efa_errno err)
 		: errno_str[err - FI_EFA_ERRNO_OFFSET];
 }
 
+static const char *efa_debug_str(enum efa_errno err)
+{
+	switch (err) {
+	case FI_EFA_REMOTE_ERROR_BAD_DEST_QPN:
+		return "This error can be caused by a crashed peer. "
+			   "Please verify the receiver status";
+	case FI_EFA_LOCAL_ERROR_UNRESP_REMOTE:
+		return "This error can be caused by a crashed peer or "
+			   "incrorrect inbound/outbound rules in the security group - "
+			   "EFA requires \"All traffic\" type allowlisting. "
+			   "Please also verify the receiver status.";
+	default:
+		return NULL;
+	}
+}
+
+const char *efa_err_conn_str(const struct efa_error_data *err_data)
+{
+	if (!err_data) {
+		return NULL;
+	}
+
+	static const char NA_STR[] = "N/A";
+	static const char format[] = "Local: %s LHost id: %s\tPeer: %s RHost id: %s";
+
+	size_t max_length = sizeof(format) - 4 +
+						MAX(HOST_ID_LENGTH + 1, sizeof(NA_STR)) * 2 +
+						MAX(sizeof(err_data->ep_raw_addr_str), sizeof(NA_STR)) +
+						MAX(sizeof(err_data->peer_raw_addr_str), sizeof(NA_STR));
```

Comment:
I think these can just be OFI_ADDRSTRLEN, since we know NA_STR < OFI_ADDRSTRLEN, and sizeof(...) is just going to return OFI_ADDRSTRLEN.